apps/search/src/lib/algolia/algoliaUtils.ts

```diff
@@ -134,7 +134,17 @@ export function productAndVariantToAlgolia({
     descriptionPlaintext: EditorJsPlaintextRenderer({ stringData: product.description }),
     slug: product.slug,
     thumbnail: product.thumbnail?.url,
+    /**
+     * @deprecated
+     */
     grossPrice: listing?.price?.amount,
+    pricing: {
+      grossAmount: variant.pricing?.price?.gross.amount,
+      onSale: variant.pricing?.onSale,
+      discountGrossAmount: variant.pricing?.discount?.gross.amount,
+      undiscountedGrossAmount: variant.pricing?.priceUndiscounted?.gross.amount,
+      channelListingPriceAmount: listing?.price?.amount,
```

Comment:
channelListingPriceAmount will be the same as the undiscounted amount

Reply:
Wonder if adding gross and net prices would be better - b2b solutions may want to display net prices, or it might be a law requirement.
I would go with direct 1:1 model of the GQL pricing object

Reply:
with gross/net -> fine

>channelListingPriceAmount will be the same as the undiscounted amount

I see in dashboard that I can only set price per channel. So how is grossPrice even calculated? What does it take?

Reply:
Channel Listings does not include discounts, but the base prices. So its gonna be the same as undiscounted price.
Pricing field should be used for getting user-visible prices